pkg/controller/actuator.go

```diff
@@ -364,112 +421,139 @@ func (a *actuator) createManagedResource(
+// triggerWebhook allows us to "reconcile" the existing EnvoyFilter which we
+// need to modify using a mutating webhook. This is achieved by sending an empty
+// patch for this EnvoyFilter, which invokes the ACL webhook component. This
+// makes sure this EnvoyFilter is kept in sync with the ACL extension config
+// (e.g. when the allowed CIDRS from the extension spec or the
+// alwaysAllowedCIDRs change).
+func (a *actuator) triggerWebhook(ctx context.Context, shootName, istioNamespace string) error {
 	// get envoyfilter with the shoot's name
-	envoyFilter := &istionetworkingClientGo.EnvoyFilter{}
+	envoyFilter := &istionetworkv1alpha3.EnvoyFilter{}
 	namespacedName := types.NamespacedName{
-		Namespace: IngressNamespace,
+		Namespace: istioNamespace,
 		Name:      shootName,
 	}
 
 	if err := a.client.Get(ctx, namespacedName, envoyFilter); err != nil {
 		return client.IgnoreNotFound(err)
 	}
 
-	if inDeletion {
+	// TODO remove migration code: previously, hash annotations have been used
+	// to check if the rule set of an ACL extension object had changed. If that
+	// was the case, the changed hash annotation would trigger the webhook.
+	// Sending an empty patch is better, as it's 1) easier and 2) also updates
+	// the EnvoyFilter when the alwaysAllowedCIDRs changed, which wasn't
+	// correctly handled before
+	// --> migration code start
+	if _, ok := envoyFilter.Annotations[HashAnnotationName]; ok {
 		delete(envoyFilter.Annotations, HashAnnotationName)
 		return a.client.Update(ctx, envoyFilter)
 	}
+	// --> migration code end
 
-	// calculate the new hash
-	newHashString, err := HashData(extSpec.Rule)
-	if err != nil {
-		return err
-	}
-
-	// get the annotation hash
-	if envoyFilter.Annotations == nil {
-		envoyFilter.Annotations = map[string]string{}
-	}
-
-	oldHashString, ok := envoyFilter.Annotations[HashAnnotationName]
-
-	// set hash if not present or different
-	if !ok || oldHashString != newHashString {
-		envoyFilter.Annotations[HashAnnotationName] = newHashString
-		return a.client.Update(ctx, envoyFilter)
-	}
-
-	// hash unchanged, do nothing
-	return nil
+	return a.client.Patch(ctx, envoyFilter, client.RawPatch(types.MergePatchType, []byte("{}")))
 }
 
 // getAllShootsWithACLExtension returns a list of all shoots that have the ACL
 // extension enabled, together with their rule.
-func (a *actuator) getAllShootsWithACLExtension(ctx context.Context) ([]envoyfilters.ACLMapping, error) {
+func (a *actuator) getAllShootsWithACLExtension(
+	ctx context.Context, istioNamespace string,
+) ([]envoyfilters.ACLMapping, map[string]string, error) {
 	extensions := &extensionsv1alpha1.ExtensionList{}
 	err := a.client.List(ctx, extensions)
 	if err != client.IgnoreNotFound(err) {
-		return nil, err
+		return nil, nil, err
 	}
 
 	if len(extensions.Items) < 1 {
-		return nil, ErrNoExtensionsFound
+		return nil, nil, ErrNoExtensionsFound
 	}
 
 	mappings := []envoyfilters.ACLMapping{}
 
+	var istioLables map[string]string
```

Comment:
```suggestion
	var istioLabels map[string]string
```

where are my Lables :D